pkg/resource/validations/kmsArnRegexValidation.go

```diff
@@ -0,0 +1,21 @@
+package validations
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var KmsArnRE = regexp.MustCompile(
+	`^arn:aws[\w-]*:kms:[\w-]+:\d{12}:key\/mrk-[0-9a-f]{32}$|[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$`,
+)
+
+func ValidateKMSKeyARN(kmsKeyARN *string) error {
+	if *kmsKeyARN == "" {
```

Comment:
Can we add a nil check on this before dereferencing it

Reply:
Also add a test case for nil strings :pray:

Reply:
We can but i don't think it'll ever be nil due to it being assigned as a string type